### PR TITLE
Capture sampling graph during warm up

### PR DESCRIPTION
**Pull request by lsy323**
# Description

Capture sampling graph during warm up. Logits is partitioned along hsize dim

# Tests

CI, offline benchmarking

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: .buildkite/pipeline_jax.yml [hunk 1]
```diff
@@ -15,6 +17,7 @@ steps:
      soft_fail: true
      env:
        MODEL_IMPL_TYPE: "vllm"
+       VLLM_XLA_CHECK_RECOMPILATION: "1"
```
Comment (xiangxu-google) [inline .buildkite/pipeline_jax.yml]: we already enabled it in https://github.com/vllm-project/tpu_commons/blob/main/.buildkite/scripts/run_in_docker.sh#L56

Comment (lsy323) [reply .buildkite/pipeline_jax.yml]: Thank you for the heads up, let me remove it. I guess ti's not expected that it's not caputrued in CI before?

**Comment by lsy323** [comment]
> Thanks a lot @lsy323 for the catch!
> 
> I have two questions:
> 
> 1. `pipeline_jax.yaml` uses `run_in_docker.sh`. `run_in_docker.sh` has `VLLM_XLA_CHECK_RECOMPILATION=1`. Then do we still need the `VLLM_XLA_CHECK_RECOMPILATION=1` additions to the yaml in this PR?
> 2. Also, I'm a bit confused, how wasn't this triggered in CI but triggered in benchmark?

Like @xiangxu-google pointed out in offline chat, CI doesn't capture this because no multi-chip in CI yet

File: .buildkite/pipeline_jax.yml (hunk 1 above)
Comment (py4) [reply .buildkite/pipeline_jax.yml]: @xiangxu-google discovered that the reason it's not triggered in CI is that CI is one single chip but benchmark is on multi chip

**Comment by lsy323** [comment]
CI Run triggered at https://buildkite.com/tpu-commons/tpu-commons-ci/builds/1213
